lib/diameter-codec.js

```diff
@@ -181,7 +181,9 @@ exports.constructResponse = function(message) {
 
 var decodeAvpHeader = function(buffer, start) {
     var avp = {};
+    if (buffer.length - start - DIAMETER_MESSAGE_AVP_CODE < 4) return undefined;
```

Comment:
I'd suggest returning `null` instead of `undefined` when the contract for a function is to return an object. `null` can be viewed as having an "invalid object" meaning.

Reply:
good idea, i'll update the PR.
also found more issues